vertical-pod-autoscaler/pkg/recommender/model/aggregate_container_state.go

```
@@ -46,6 +47,22 @@ import (
 	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/recommender/util"
 )
 
+var (
+	globalPruningGracePeriodDuration = flag.String("pruning-grace-period-duration", "", `The grace period for deleting stale aggregates and recommendations. An empty duration will disable the grace period for all containers by default.`)
+	parsedPruningGracePeriodDuration = parsePruningGracePeriodDuration()
+)
+
+func parsePruningGracePeriodDuration() *time.Duration {
+	if globalPruningGracePeriodDuration == nil || *globalPruningGracePeriodDuration == "" {
+		return nil
+	}
+	duration, err := time.ParseDuration(*globalPruningGracePeriodDuration)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to parse --pruning-grace-period-duration: %v", err))
```

Instead of using `panic()`, we should return an error to be handled by the caller of this function. Could we modify this to return an error using `klog.ErrorS()` followed by `os.Exit(255)`?
What do you think?

---

Fixed in https://github.com/kubernetes/autoscaler/pull/6745/commits/b2ae65af20c3bb75c5475deecf1241f523946531, does it make sense? I used `klog.Fatalf` instead of error and exit.

---

Yes, that's fine, but could you use `Errors` and `Exit`? It's the correct approach for structural logging.

---

Hopefully https://github.com/kubernetes/autoscaler/pull/6745/commits/cd63322b24f38c9f5a3b896d5c6bf03c062f1111 is correct? :crossed_fingers:

---

Yup! Thanks